### PR TITLE
[3268] Date-time-picker inside the targeting dialog.

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/3268

- Added timezone selector
- Set clear on date-time selector
- Close dialog on click targeting dialog (when opened)
- Minor fixes

### CODE REVIEW

File: static-assets/components/cstudio-contextual-nav/targeting.js [hunk 1]
```diff
@@ -15,12 +15,17 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+CStudioAuthoring.Utils.addJavascript("/static-assets/jquery/timezones/timezones.full.js");
```
Comment (rart) [inline static-assets/components/cstudio-contextual-nav/targeting.js]: add conditionally, only if plugin is not already loaded

File: static-assets/components/cstudio-contextual-nav/targeting.js [hunk 2]
```diff
@@ -308,14 +321,17 @@ CStudioAuthoring.ContextualNav.TargetingMod = CStudioAuthoring.ContextualNav.Tar
                       break;
                     case "datetime":
                       var $controlEl = $(controlEl),
-                          $dateTimePicker = $controlEl.find('.date-picker');
+                          $dateTimePicker = $controlEl.find('.date-picker'),
+                          $timeZonePicker = $controlEl.siblings('.timezone-container').find('.zone-picker');
 
                       if (currentProp.default_value === '') {
                         $dateTimePicker.val(currentProp.default_value);
+                        $timeZonePicker.hide();
```
Comment (rart) [inline static-assets/components/cstudio-contextual-nav/targeting.js]: why hidden here? what’s the logic?

**Review by rart — CHANGES_REQUESTED** [review]

File: static-assets/components/cstudio-contextual-nav/targeting.js (hunk 2 above)
Comment (jvega190) [reply static-assets/components/cstudio-contextual-nav/targeting.js]: When datetime selector is cleared (input empty) it hides the timezone selector as it is not needed (you can save empty value on the datetime selector)

Comment (rart) [reply static-assets/components/cstudio-contextual-nav/targeting.js]: Let's leave it visible - just like the timepicker is all the time.

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**